src/internal_modules/roc_audio/depacketizer.cpp

```diff
@@ -107,21 +112,32 @@ Depacketizer::read_samples_(sample_t* buff_ptr, sample_t* buff_end, FrameInfo& i
                 * (size_t)packet::timestamp_diff(next_timestamp, timestamp_);
 
             const size_t max_samples = (size_t)(buff_end - buff_ptr);
+            const size_t nsamples = std::min(mis_samples, max_samples);
 
-            buff_ptr = read_missing_samples_(
-                buff_ptr, buff_ptr + std::min(mis_samples, max_samples));
+            buff_ptr = read_missing_samples_(buff_ptr, buff_ptr + nsamples);
+            if (!info.ts) {
+                info.ts =
+                    next_capture_ts_ - sample_spec_.samples_overall_2_ns(mis_samples);
+            }
         }
 
         if (buff_ptr < buff_end) {
             sample_t* new_buff_ptr = read_packet_samples_(buff_ptr, buff_end);
+            const size_t nsamples = size_t(new_buff_ptr - buff_ptr);
 
-            info.n_decoded_samples += size_t(new_buff_ptr - buff_ptr);
-
+            info.n_decoded_samples += nsamples;
+            if (nsamples && !info.ts) {
+                info.ts = next_capture_ts_;
+            }
+            next_capture_ts_ += sample_spec_.samples_overall_2_ns(nsamples);
```

Comment:
I guess we should do it only if valid_capture_ts_ is true.

BTW I think we need a test for depacketizer when timestamps in packets are zero for a while and then become non-zero, to catch this gotchas.